packages/templates/clients/js/websocket/template/client.js.js

```diff
@@ -1,109 +1,38 @@
   return (
     <File name={params.clientFileName}>
-      <FileHeaderInfo
-        info={info}
-        server={server}
-      />
-      <Requires/>
+      <FileHeaderInfo info={info} server={server} />
+      <Requires />
       <Text>
-        {`class ${getClientName(info)} {
-  constructor() {
-    this.url = '${getServerUrl(server)}';
-    this.websocket = null;
-    this.messageHandlers = [];
-    this.errorHandlers = [];
-  }
-
-  // Method to establish a WebSocket connection
-  connect() {
-    return new Promise((resolve, reject) => {
-      this.websocket = new WebSocket(this.url);
-
-      // On successful connection
-      this.websocket.onopen = () => {
-        console.log('Connected to ${title} server');
-        resolve();
-      };
-
-      // On receiving a message
-      this.websocket.onmessage = (event) => {
-        console.log('Message received:', event.data);
-
-        this.messageHandlers.forEach(handler => {
-          if (typeof handler === 'function') {
-            this.handleMessage(event.data, handler);
-          }
-        });
-      };
-
-      // On error first call custom error handlers, then default error behavior
-      this.websocket.onerror = (error) => {
-        if (this.errorHandlers.length > 0) {
-          // Call custom error handlers
-          this.errorHandlers.forEach(handler => handler(error));
-        } else {
-          // Default error behavior
-          console.error('WebSocket Error:', error);
-        }
-        reject(error);
-      };
-
-      // On connection close
-      this.websocket.onclose = () => {
-        console.log('Disconnected from ${title} server');
-      };
-    });
-  }
-
-  // Method to register custom error handlers
-  registerMessageHandler(handler) {
-    if (typeof handler === 'function') {
-      this.messageHandlers.push(handler);
-    } else {
-      console.warn('Message handler must be a function');
-    }
-  }
-
-  // Method to register custom error handlers
-  registerErrorHandler(handler) {
-    if (typeof handler === 'function') {
-      this.errorHandlers.push(handler);
-    } else {
-      console.warn('Error handler must be a function');
-    }
-  }
-
-  // Method to handle message with callback
-  handleMessage(message, cb) {
-    if (cb) cb(message);
-  }
+        {`
+${ClassDeclaration({ clientName, serverUrl })}
 
-  // Method to send an echo message to the server
-  sendEchoMessage(message) {
-    this.websocket.send(JSON.stringify(message));
-    console.log('Sent message to echo server:', message);
-  }
+${ConnectMethod({ title: info.title() })}
 
-  // Method to close the WebSocket connection
-  close() {
-    if (this.websocket) {
-      this.websocket.close();
-      console.log('WebSocket connection closed.');
-    }
-  }
-}
+${HandlerRegistration({ handlerType: 'Message' })}
+${HandlerRegistration({ handlerType: 'Error' })}
 
-module.exports = ${getClientName(info)};`}
+${HandleMessageMethod()}
+${SendEchoMessageMethod()}
+${CloseMethod()}
```

Comment:
any reason why you did not use these new components as components but as functions? like for example `<ModuleExport clientName={clientName} />`